### ISSUE TITLE
Setting UPSname? Initial change of password and reboot prompt

### ISSUE THREAD

**Issue by irrwitzer42** [haught/ansible_apcos#1]
Hi Matt,

first of all: thanks a BUNCH for your effort!
I do have the unthankful job of rolling out a lot UPSes at the moment and thanks to your work it's almost a pleasure doing so.

I do have a question, that is not directly about your ansible role, but I am hoping you might have an idea for me, so please excuse the wrong channel for it.

### I've been looking for a way to set the UPSname, which is then exposed via SNMP as upsIdentName and upsBasicIdentName. 

It's default is "APC UPS" and it can be set in the webUI under configuration -> UPS -> "UPS name"
I'm already setting:

```
commands:                             
  - "system -n {{inventory_hostname}}"
  - "tcpip -h {{inventory_hostname}}" 
  - "dns -h {{inventory_hostname}}"   

```
but none of those get synced to this UPS name. Any chance you know how to set it? APC's CLI documentation isn't helpful here - and neither is their support.

And while writing I remembered two other issues I got:

### 1) Your example code for interactive commands and rebooting:

```
- name: Reboot UPS to activate changes                        
  haught.apcos.apcos_command:                                 
    commands:                                                 
      - command: 'reboot'                                     
        prompt: "Enter 'YES' to continue or <ENTER> to cancel"
        answer: "YES"                                         

```
doesn't work as expected here:

```
__rpc__\nansible.module_util
s.connection.ConnectionError: {'command': 'reboot', 'prompt': \"Enter 'YES' to continue or <ENTER> to cancel\", 'answer': 'YES'}\r\nE101: Command Not Found\r\n\r\n* Reboot required for change to take effect.\r\napc>\n"

```
The coresponding dialog on the CLI looks like this:

```
* Reboot required for change to take effect.
apc>reboot
E000: Success
Reboot Management Interface
Enter 'YES' to continue or ENTER to cancel: YES
Rebooting...



* Reboot required for change to take effect.
apc>

```

It is rebooting sometimes (despite the errors), and sometimes not.

### 2) When initially deploying a new UPS (or a factory resetted one) the NMC requires to set a new password for the apc user.
Is there a way to script that, too? Since there is no command I can send first (because it's asking for a new password directly after establishing the ssh session) I can't use the snippets above. Did you find a solution for this? Otherwise I will do this by hand - like I have to upgrade the firmware manually.

Thanks again for your great work! 

Best regards,
   Johannes

**Comment by haught** [comment]
Thanks Johannes for using the module, I always wondered if anyone would find it useful.

For UPSname, I have no idea how to set this via the CLI. We only use the sysName at my work. I can look around, but it simply might not be a feature they have added. I am wanting a way to disable the alarm via the CLI...

As for 1), I have made a few changes in the most recent release. It should be working properly.

As for 2), What I did was write some python scripts for cron to: 
  1. find our dhcp free devices in InfoBlox matching the APC MAC prefix in a few VLANs and flip their type to a reserved dhcp
  2. use pexpect to set the initial password
  3. enable FTP, send over the firmware, reboot, wait, disable FTP
  4. kick of AWX job to configure everything

I was never able to get scp'ing the firmware to work at all, had to use FTP. Haven't tried the latest firmware.

I have been wanting to get this all in ansible, just haven't had the time. Might be able to use ansible.builtin.raw without gather_facts to shotgun the initial pw change.
